### PR TITLE
pulling OpenShiftVersion call/info into own struct & funcs

### PR DESCRIPTION
**Pull request by jeremyary**
New OpenShiftVersion struct split out of PlatformInfo. New struct is populated as a separate utils func now. This method will approximate OCP version by mapped k8s version, then attempt to refine version info via API calls if possible. Doing both should prevent user permissions issues from causing a complete failure to resolve a version.

### CODE REVIEW

**Review by bmozaffa — CHANGES_REQUESTED** [review]
@jeremyary I have trouble following exactly how this works and how it should be used, and don't want to hold you up, but also want to make sure it's clear enough for users to want to leverage it.

I think you are trying too hard to use OO design in a language that's not made for it. Based on what you have done so far and to avoid multiple backend calls, I would like to see `utils.go` have the following:

```
func IsOpenShift(cfg *rest.Config) (bool, error)
func GetPlatformInfo(cfg *rest.Config) (platform.PlatformInfo, error)
func LookupOpenShiftVersion(cfg *rest.Config) (platform.OpenShiftVersion, error)
func MapKnownVersion(info PlatformInfo) platform.OpenShiftVersion
```

**Comment by jeremyary** [comment]
> I would like to see `utils.go` have the following:

ok, see most recent changes

**Comment by jeremyary** [comment]
Platform_versioner and all types have been moved to internal/platform directory. Utils funcs rearranged and some parameters/logic simplified a bit to reflect helper method usage.

File: pkg/utils/openshift/utils.go [hunk 1]
```diff
@@ -1,37 +1,53 @@
...
-LookupOpenShiftVersion refines PlatformInfo.OCPVersion info based on API calls, rather than deriving from K8SVersion.
+LookupOpenShiftVersion fetches OpenShift version info from API endpoints
 *** NOTE: OCP 4.1+ requires elevated user permissions, see PlatformVersioner for details
 Accepts <nil> or instantiated 'cfg' rest config parameter.
-Accepts prepopulated or empty PlatformInfo{}
 
-Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1.2, K8SVersion: 1.13+, OS: linux/amd64 }
+Result: OpenShiftVersion{ Version: 4.1.2 }
+*/
+func LookupOpenShiftVersion(info platform.PlatformInfo) (platform.OpenShiftVersion, error) {
```
Comment (bmozaffa) [inline pkg/utils/openshift/utils.go]: Without doing a deep dive, I'm surprised you don't need `cfg *rest.Config` here?!

Comment (jeremyary) [reply pkg/utils/openshift/utils.go]: You're right, I factored part of it away when there was still some backend overlap with PlatformInfo, but they share no logic now and each need cfg to make their own calls. Forgot to swap it back out - updated now.

**Review by bmozaffa — APPROVED** [review]

**Pull request merged**